pkg/ipam/poolallocator/allocator.go

```diff
@@ -320,6 +377,50 @@ func (a *IPPoolAllocator) ReleasePod(namespace, podName string) error {
 	return err
 }
 
+// ReleaseResource releases all IPs associated with specified StatefulSet. It returns error
+// in case CRD failed to update its state or if pool or StatefulSet are not present.
+// In case of success, IP pool CRD status is updated with released entries.
+func (a *IPPoolAllocator) ReleaseStatefulSet(namespace, name string) error {
```

Comment:
For antrea-controller to support StatefulSet, I think we only need this function. Can we use this PR for the implementation on antrea-controller?
The PR for antrea-agent side is #3042

Reply:
I was planning a separate PR but can dump it here if you prefer.
Allocation code is indeed aimed at next phase, but since it provides consistency and helps with UTs here I'd prefer to merge it as well as part of this PR.

Reply:
Added antrea controller support. This PR will need to be refactored on top of https://github.com/antrea-io/antrea/pull/3046 and will need to accommodate `Preallocated` -> `Reserved` name change if it will take place.